src/dashboard/tests/test_administration_storage.py

```diff
@@ -0,0 +1,150 @@
+                "pipeline": ["/api/v2/pipeline/fe263021-f1d7-4a25-b691-df80da5ee048/"],
+                "used": "0",
+                "description": "Store DIP in standard Archivematica Directory",
+                "space": "/api/v2/space/85a6a5af-8b99-4e04-83da-4d1d712f1115/",
+                "enabled": True,
+                "quota": None,
+                "relative_path": "var/archivematica/sharedDirectory/www/DIPsStore",
+                "purpose": "DS",
+                "path": "/var/archivematica/sharedDirectory/www/DIPsStore",
+                "resource_uri": "/api/v2/location/6e4cf229-e614-436d-9055-839dfe3145a6/",
+            },
+            {
+                "uuid": "8ace39fe-d52e-4b84-a8ab-3c68fdfea829",
+                "pipeline": ["/api/v2/pipeline/fe263021-f1d7-4a25-b691-df80da5ee048/"],
+                "used": "0",
+                "description": "Default transfer backlog",
+                "space": "/api/v2/space/85a6a5af-8b99-4e04-83da-4d1d712f1115/",
+                "enabled": True,
+                "quota": None,
+                "relative_path": "var/archivematica/sharedDirectory/www/AIPsStore/transferBacklog",
+                "purpose": "BL",
+                "path": "/var/archivematica/sharedDirectory/www/AIPsStore/transferBacklog",
+                "resource_uri": "/api/v2/location/8ace39fe-d52e-4b84-a8ab-3c68fdfea829/",
+            },
+            {
+                "uuid": "b3333b2a-5f3d-4c32-86b4-d334ff80c111",
+                "pipeline": ["/api/v2/pipeline/fe263021-f1d7-4a25-b691-df80da5ee048/"],
+                "used": "0",
+                "description": "Default AIP recovery",
+                "space": "/api/v2/space/85a6a5af-8b99-4e04-83da-4d1d712f1115/",
+                "enabled": True,
+                "quota": None,
+                "relative_path": "var/archivematica/storage_service/recover",
+                "purpose": "AR",
+                "path": "/var/archivematica/storage_service/recover",
+                "resource_uri": "/api/v2/location/b3333b2a-5f3d-4c32-86b4-d334ff80c111/",
+            },
+        ]
+        response = self.client.get(self.url)
+        locations = response.context["locations"]
+        # The currently processing and AIP recovery locations are removed
+        self.assertFalse([loc for loc in locations if loc["purpose"] == "CP"])
+        self.assertFalse([loc for loc in locations if loc["purpose"] == "AR"])
+        # Disabled location is removed
+        self.assertFalse(
+            [
+                loc
+                for loc in locations
+                if loc["uuid"] == "1232e1d5-a67a-4791-b4a9-4359072747bf"
+            ]
+        )
+        # Only two locations show usage
+        self.assertEqual(len([loc for loc in locations if loc["show_usage"]]), 2)
+        # One of them has an unlimited quota set
+        self.assertEqual(
+            len([loc for loc in locations if loc["quota"] == "unlimited"]), 1
+        )
```

Comment:
This is an aside... (it works well in the UI!): what does the quota setting actually do in the storage service? Does it trigger any actions?

Reply:
It's being checked/updated when packages are moved from/to a SS location and it should cause an error if it's reached in that process, but I think it still needs some enhancements. There are some minor comments about it in https://github.com/archivematica/Issues/issues/406.